src/lib.rs

```diff
@@ -395,13 +395,20 @@ where
 
         remaining = largest_id - start_id + 1;
 
+        // Find the percent value rounded to the thousand-th of a percent.
+        let percent_complete =
```

Comment:
Just wondering if you might want to check for remaining is > largest_id, leading to negative percent?

Reply:
added some asserts.